File: 프로그래머스/2/389479. 서버 증설 횟수/서버 증설 횟수.py
# 서버 증설 조건 m
# 서버 운영시간 k

def solution(players, m, k):
    server = []
    # Player식
    # 특정 이용자수가 i 일 때 필요한 서버의 개수를 구하는 식
    # i//m 
    answer = 0
    for i in players:
        server_cnt = len(server)
        print(f'server:{server}, i:{i}, ')
        if i//m > server_cnt:
            for j in range(i//m - server_cnt):
                server.append(k)
                answer +=1
        idx = 0
        while idx < len(server):
            server[idx] -= 1
            if server[idx] == 0:
                server.pop(idx)
            else:
                idx += 1
    print(server)
    return answer

File: 프로그래머스/2/389479. 서버 증설 횟수/서버 증설 횟수.java
import java.util.*;
class Solution {
    public int solution(int[] players, int m, int k) {
        int [] servers = new int[players.length];
        int serverCnt = 0;
        for (int ser: servers){
            System.out.println(ser);
        }
        for (int i = 0; i < players.length; i++) {
            int need = players[i] / m; 
            int curServer = 0;
            
            for (int j = i; j >= Math.max(i-k+1, 0); j--) {
                curServer += servers[j];
                // System.out.println(j+"번째 서버: "+servers[j]);
            }
            // System.out.println("필요한 서버수: "+need+" curServer: "+curServer);
            if (curServer < need) {
                int toAdd = need - curServer;
                // System.out.println("toAdd"+toAdd);
                servers[i] = toAdd;
                serverCnt += toAdd;
            }
        }
        return serverCnt;
    }
}